notebooks/dynamic_tutorial.py

```diff
@@ -0,0 +1,287 @@
+# According to the video tutorial here: https://youtu.be/oBt53YbR9Kk
+
+#%%
+# Fibonacci numbers, brute force approach
+
+def fib_brute(n):
+    if n <= 2:
+        return 1
+    else:
+        return fib_brute(n - 1) + fib_brute(n - 2)
+
+# to evaluate complexity it's useful to draw a solution graph, in this case tree
+"""
+O(2^n) time complexity
+O(n) space complexity
+"""
+
+fib_brute(9)
+
+#%%
+# Fibonacci numbers, memoization
+
+def fib(n, memo={}):
```

Comment:
this is nice usage of memoisation,
checkout `lru_cache` decorator from functools
@djfedos